arch/src/x86_64/mod.rs

```diff
@@ -675,6 +675,25 @@ pub fn generate_common_cpuid(
         });
     }
 
+    // Check if KVM provides 0x8000_0006 (L2 cache information - added Linux 5.7) else
+    // fetch host CPUID version if available. This is not as good as the KVM version
+    // due to handling of hybrid CPUs.
+    if !cpuid.iter().any(|c| c.function == 0x8000_0006)
```

Comment:
Don't you need to delete the old CpuIdEntry here? It is got from KVM_GET_SUPPORTED_CPUID.